services/wireguard/endpoint/kernelspace/client.go

```diff
@@ -84,20 +92,46 @@ func (c *client) AddPeer(iface string, peer wg.PeerInfo) error {
 
 func (c *client) up(iface string, ipAddr net.IPNet) error {
 	if d, err := c.wgClient.Device(iface); err != nil || d.Name != iface {
-		if err := exec.Command("ip", "link", "add", "dev", iface, "type", "wireguard").Run(); err != nil {
+		if err := ipExec("link", "add", "dev", iface, "type", "wireguard"); err != nil {
 			return err
 		}
 	}
 
-	if err := exec.Command("ip", "address", "replace", "dev", iface, ipAddr.String()).Run(); err != nil {
+	if err := ipExec("address", "replace", "dev", iface, ipAddr.String()); err != nil {
 		return err
 	}
 
-	return exec.Command("ip", "link", "set", "dev", iface, "up").Run()
+	return ipExec("link", "set", "dev", iface, "up")
+}
+
+func (c *client) addRoute(iface string, endpoint net.UDPAddr) error {
+	gw, err := gateway.DiscoverGateway()
+	if err != nil {
+		return err
+	}
+
+	if err := ipExec("route", "replace", endpoint.IP.String(), "via", gw.String()); err != nil {
+		return err
+	}
+	if err := ipExec("route", "replace", "0.0.0.0/1", "dev", iface); err != nil {
+		return err
+	}
+	return ipExec("route", "replace", "128.0.0.0/1", "dev", iface)
 }
 
 func (c *client) Close() error {
-	if err := exec.Command("ip", "link", "del", "dev", c.iface).Run(); err != nil {
+	d, err := c.wgClient.Device(c.iface)
+	if err != nil || d.Name != c.iface {
+		return err
+	}
+
+	if len(d.Peers) > 0 && d.Peers[0].Endpoint != nil {
+		if err := ipExec("route", "del", d.Peers[0].Endpoint.IP.String()); err != nil {
```

Comment:
We should not use `route` command on linux since we are already using `ip`. Route is completely covered by `ip route` subcommand.

Reply:
`ipExec` is calling `ip` tool inside, all function args are arguments for `ip` tool. 
Looks like it is a confusing syntax, and it's not clear.